src/ucp/wireup/address.c

```diff
@@ -312,16 +319,66 @@ static void ucp_address_memcheck(ucp_context_h context, void *ptr, size_t size,
     }
 }
 
+static unsigned ucp_address_pack_flags(uint64_t packed_flags,
+                                       uint64_t cap_flags,
+                                       unsigned start_bit,
+                                       uint32_t *result_flags)
+{
+    uint64_t bit          = 1;
+    uint32_t packed_flag  = UCS_BIT(start_bit);
+    unsigned num_cap_bits = 0;
+
+    while (cap_flags & ~(bit - 1)) {
```

Comment:
not sure if i completely understand pack/unpack algo but can we do the same just using bit ops without the loop?

Reply:
done